// Copyright (c) 2020-2022 The Open-Transactions developers
// This Source Code Form is subject to the terms of the Mozilla Public
// License, v. 2.0. If a copy of the MPL was not distributed with this
// file, You can obtain one at http://mozilla.org/MPL/2.0/.

syntax = "proto2";

package opentxs.proto;
option java_package = "org.opentransactions.proto";
option java_outer_classname = "OTContactEnums";
option optimize_for = LITE_RUNTIME;

enum ContactSectionName {
    CONTACTSECTION_ERROR = 0;
    CONTACTSECTION_SCOPE = 1;
    CONTACTSECTION_IDENTIFIER = 2;
    CONTACTSECTION_ADDRESS = 3;
    CONTACTSECTION_COMMUNICATION = 4;
    CONTACTSECTION_PROFILE = 5;
    CONTACTSECTION_RELATIONSHIP = 6;
    CONTACTSECTION_DESCRIPTOR = 7;
    CONTACTSECTION_EVENT = 8;
    CONTACTSECTION_CONTRACT = 9;
    CONTACTSECTION_PROCEDURE = 10;
}

enum ContactItemType {
    CITEMTYPE_ERROR = 0;
    CITEMTYPE_INDIVIDUAL = 1;
    CITEMTYPE_ORGANIZATION = 2;
    CITEMTYPE_BUSINESS = 3;
    CITEMTYPE_GOVERNMENT = 4;
    CITEMTYPE_SERVER = 5;
    CITEMTYPE_PREFIX = 6;
    CITEMTYPE_FORENAME = 7;
    CITEMTYPE_MIDDLENAME = 8;
    CITEMTYPE_SURNAME = 9;
    CITEMTYPE_PEDIGREE = 10;
    CITEMTYPE_SUFFIX = 11;
    CITEMTYPE_NICKNAME = 12;
    CITEMTYPE_COMMONNAME = 13;
    CITEMTYPE_PASSPORT = 14;
    CITEMTYPE_NATIONAL = 15;
    CITEMTYPE_PROVINCIAL = 16;
    CITEMTYPE_MILITARY = 17;
    CITEMTYPE_PGP = 18;
    CITEMTYPE_OTR = 19;
    CITEMTYPE_SSL = 20;
    CITEMTYPE_PHYSICAL = 21;
    CITEMTYPE_OFFICIAL = 22;
    CITEMTYPE_BIRTHPLACE = 23;
    CITEMTYPE_HOME = 24;
    CITEMTYPE_WEBSITE = 25;
    CITEMTYPE_OPENTXS = 26;
    CITEMTYPE_PHONE = 27;
    CITEMTYPE_EMAIL = 28;
    CITEMTYPE_SKYPE = 29;
    CITEMTYPE_WIRE = 30;
    CITEMTYPE_QQ = 31;
    CITEMTYPE_BITMESSAGE = 32;
    CITEMTYPE_WHATSAPP = 33;
    CITEMTYPE_TELEGRAM = 34;
    CITEMTYPE_KIK = 35;
    CITEMTYPE_BBM = 36;
    CITEMTYPE_WECHAT = 37;
    CITEMTYPE_KAKAOTALK = 38;
    CITEMTYPE_FACEBOOK = 39;
    CITEMTYPE_GOOGLE = 40;
    CITEMTYPE_LINKEDIN = 41;
    CITEMTYPE_VK = 42;
    CITEMTYPE_ABOUTME = 43;
    CITEMTYPE_ONENAME = 44;
    CITEMTYPE_TWITTER = 45;
    CITEMTYPE_MEDIUM = 46;
    CITEMTYPE_TUMBLR = 47;
    CITEMTYPE_YAHOO = 48;
    CITEMTYPE_MYSPACE = 49;
    CITEMTYPE_MEETUP = 50;
    CITEMTYPE_REDDIT = 51;
    CITEMTYPE_HACKERNEWS = 52;
    CITEMTYPE_WIKIPEDIA = 53;
    CITEMTYPE_ANGELLIST = 54;
    CITEMTYPE_GITHUB = 55;
    CITEMTYPE_BITBUCKET = 56;
    CITEMTYPE_YOUTUBE = 57;
    CITEMTYPE_VIMEO = 58;
    CITEMTYPE_TWITCH = 59;
    CITEMTYPE_SNAPCHAT = 60;
    CITEMTYPE_VINE = 61;
    CITEMTYPE_INSTAGRAM = 62;
    CITEMTYPE_PINTEREST = 63;
    CITEMTYPE_IMGUR = 64;
    CITEMTYPE_FLICKR = 65;
    CITEMTYPE_DRIBBLE = 66;
    CITEMTYPE_BEHANCE = 67;
    CITEMTYPE_DEVIANTART = 68;
    CITEMTYPE_SPOTIFY = 69;
    CITEMTYPE_ITUNES = 70;
    CITEMTYPE_SOUNDCLOUD = 71;
    CITEMTYPE_ASKFM = 72;
    CITEMTYPE_EBAY = 73;
    CITEMTYPE_ETSY = 74;
    CITEMTYPE_OPENBAZAAR = 75;
    CITEMTYPE_XBOXLIVE = 76;
    CITEMTYPE_PLAYSTATION = 77;
    CITEMTYPE_SECONDLIFE = 78;
    CITEMTYPE_WARCRAFT = 79;
    CITEMTYPE_ALIAS = 80;
    CITEMTYPE_ACQUAINTANCE = 81;
    CITEMTYPE_FRIEND = 82;
    CITEMTYPE_SPOUSE = 83;
    CITEMTYPE_SIBLING = 84;
    CITEMTYPE_MEMBER = 85;
    CITEMTYPE_COLLEAGUE = 86;
    CITEMTYPE_PARENT = 87;
    CITEMTYPE_CHILD = 88;
    CITEMTYPE_EMPLOYER = 89;
    CITEMTYPE_EMPLOYEE = 90;
    CITEMTYPE_CITIZEN = 91;
    CITEMTYPE_PHOTO = 92;
    CITEMTYPE_GENDER = 93;
    CITEMTYPE_HEIGHT = 94;
    CITEMTYPE_WEIGHT = 95;
    CITEMTYPE_HAIR = 96;
    CITEMTYPE_EYE = 97;
    CITEMTYPE_SKIN = 98;
    CITEMTYPE_ETHNICITY = 99;
    CITEMTYPE_LANGUAGE = 100;
    CITEMTYPE_DEGREE = 101;
    CITEMTYPE_CERTIFICATION = 102;
    CITEMTYPE_TITLE = 103;
    CITEMTYPE_SKILL = 104;
    CITEMTYPE_AWARD = 105;
    CITEMTYPE_LIKES = 106;
    CITEMTYPE_SEXUAL = 107;
    CITEMTYPE_POLITICAL = 108;
    CITEMTYPE_RELIGIOUS = 109;
    CITEMTYPE_BIRTH = 110;
    CITEMTYPE_SECONDARYGRADUATION = 111;
    CITEMTYPE_UNIVERSITYGRADUATION = 112;
    CITEMTYPE_WEDDING = 113;
    CITEMTYPE_ACCOMPLISHMENT = 114;
    CITEMTYPE_BTC = 115;
    CITEMTYPE_ETH = 116;
    CITEMTYPE_XRP = 117;
    CITEMTYPE_LTC = 118;
    CITEMTYPE_DAO = 119;
    CITEMTYPE_XEM = 120;
    CITEMTYPE_DASH = 121;
    CITEMTYPE_MAID = 122;
    CITEMTYPE_LSK = 123;
    CITEMTYPE_DOGE = 124;
    CITEMTYPE_DGD = 125;
    CITEMTYPE_XMR = 126;
    CITEMTYPE_WAVES = 127;
    CITEMTYPE_NXT = 128;
    CITEMTYPE_SC = 129;
    CITEMTYPE_STEEM = 130;
    CITEMTYPE_AMP = 131;
    CITEMTYPE_XLM = 132;
    CITEMTYPE_FCT = 133;
    CITEMTYPE_BTS = 134;
    CITEMTYPE_USD = 135;
    CITEMTYPE_EUR = 136;
    CITEMTYPE_GBP = 137;
    CITEMTYPE_INR = 138;
    CITEMTYPE_AUD = 139;
    CITEMTYPE_CAD = 140;
    CITEMTYPE_SGD = 141;
    CITEMTYPE_CHF = 142;
    CITEMTYPE_MYR = 143;
    CITEMTYPE_JPY = 144;
    CITEMTYPE_CNY = 145;
    CITEMTYPE_NZD = 146;
    CITEMTYPE_THB = 147;
    CITEMTYPE_HUF = 148;
    CITEMTYPE_AED = 149;
    CITEMTYPE_HKD = 150;
    CITEMTYPE_MXN = 151;
    CITEMTYPE_ZAR = 152;
    CITEMTYPE_PHP = 153;
    CITEMTYPE_SEK = 154;
    CITEMTYPE_TNBTC = 155;
    CITEMTYPE_TNXRP = 156;
    CITEMTYPE_TNLTC = 157;
    CITEMTYPE_TNXEM = 158;
    CITEMTYPE_TNDASH = 159;
    CITEMTYPE_TNMAID = 160;
    CITEMTYPE_TNLSK = 161;
    CITEMTYPE_TNDOGE = 162;
    CITEMTYPE_TNXMR = 163;
    CITEMTYPE_TNWAVES = 164;
    CITEMTYPE_TNNXT = 165;
    CITEMTYPE_TNSC = 166;
    CITEMTYPE_TNSTEEM = 167;
    CITEMTYPE_PHILOSOPHY = 168;
    CITEMTYPE_MET = 169;
    CITEMTYPE_FAN = 170;
    CITEMTYPE_SUPERVISOR = 171;
    CITEMTYPE_SUBORDINATE = 172;
    CITEMTYPE_CONTACT = 173;
    CITEMTYPE_REFRESHED = 174;
    CITEMTYPE_BOT = 175;
    CITEMTYPE_BCH = 176;
    CITEMTYPE_TNBCH = 177;
    CITEMTYPE_OWNER = 178;
    CITEMTYPE_PROPERTY = 179;
    CITEMTYPE_UNKNOWN = 180;
    CITEMTYPE_ETHEREUM_OLYMPIC = 181;
    CITEMTYPE_ETHEREUM_CLASSIC = 182;
    CITEMTYPE_ETHEREUM_EXPANSE = 183;
    CITEMTYPE_ETHEREUM_MORDEN = 184;
    CITEMTYPE_ETHEREUM_ROPSTEN = 185;
    CITEMTYPE_ETHEREUM_RINKEBY = 186;
    CITEMTYPE_ETHEREUM_KOVAN = 187;
    CITEMTYPE_ETHEREUM_SOKOL = 188;
    CITEMTYPE_ETHEREUM_POA = 189;
    CITEMTYPE_PKT = 190;
    CITEMTYPE_TNPKT = 191;
    CITEMTYPE_REGTEST = 192;
    CITEMTYPE_BNB = 193;
    CITEMTYPE_SOL = 194;
    CITEMTYPE_USDT = 195;
    CITEMTYPE_ADA = 196;
    CITEMTYPE_DOT = 197;
    CITEMTYPE_USDC = 198;
    CITEMTYPE_SHIB = 199;
    CITEMTYPE_LUNA = 200;
    CITEMTYPE_AVAX = 201;
    CITEMTYPE_UNI = 202;
    CITEMTYPE_LINK = 203;
    CITEMTYPE_WBTC = 204;
    CITEMTYPE_BUSD = 205;
    CITEMTYPE_MATIC = 206;
    CITEMTYPE_ALGO = 207;
    CITEMTYPE_VET = 208;
    CITEMTYPE_AXS = 209;
    CITEMTYPE_ICP = 210;
    CITEMTYPE_CRO = 211;
    CITEMTYPE_ATOM = 212;
    CITEMTYPE_THETA = 213;
    CITEMTYPE_FIL = 214;
    CITEMTYPE_TRX = 215;
    CITEMTYPE_FTT = 216;
    CITEMTYPE_ETC = 217;
    CITEMTYPE_FTM = 218;
    CITEMTYPE_DAI = 219;
    CITEMTYPE_BTCB = 220;
    CITEMTYPE_EGLD = 221;
    CITEMTYPE_HBAR = 222;
    CITEMTYPE_XTZ = 223;
    CITEMTYPE_MANA = 224;
    CITEMTYPE_NEAR = 225;
    CITEMTYPE_GRT = 226;
    CITEMTYPE_CAKE = 227;
    CITEMTYPE_EOS = 228;
    CITEMTYPE_FLOW = 229;
    CITEMTYPE_AAVE = 230;
    CITEMTYPE_KLAY = 231;
    CITEMTYPE_KSM = 232;
    CITEMTYPE_XEC = 233;
    CITEMTYPE_MIOTA = 234;
    CITEMTYPE_HNT = 235;
    CITEMTYPE_RUNE = 236;
    CITEMTYPE_BSV = 237;
    CITEMTYPE_LEO = 238;
    CITEMTYPE_NEO = 239;
    CITEMTYPE_ONE = 240;
    CITEMTYPE_QNT = 241;
    CITEMTYPE_UST = 242;
    CITEMTYPE_MKR = 243;
    CITEMTYPE_ENJ = 244;
    CITEMTYPE_CHZ = 245;
    CITEMTYPE_AR = 246;
    CITEMTYPE_STX = 247;
    CITEMTYPE_BTT = 248;
    CITEMTYPE_HOT = 249;
    CITEMTYPE_SAND = 250;
    CITEMTYPE_OMG = 251;
    CITEMTYPE_CELO = 252;
    CITEMTYPE_ZEC = 253;
    CITEMTYPE_COMP = 254;
    CITEMTYPE_TFUEL = 255;
    CITEMTYPE_KDA = 256;
    CITEMTYPE_LRC = 257;
    CITEMTYPE_QTUM = 258;
    CITEMTYPE_CRV = 259;
    CITEMTYPE_HT = 260;
    CITEMTYPE_NEXO = 261;
    CITEMTYPE_SUSHI = 262;
    CITEMTYPE_KCS = 263;
    CITEMTYPE_BAT = 264;
    CITEMTYPE_OKB = 265;
    CITEMTYPE_DCR = 266;
    CITEMTYPE_ICX = 267;
    CITEMTYPE_RVN = 268;
    CITEMTYPE_SCRT = 269;
    CITEMTYPE_REV = 270;
    CITEMTYPE_AUDIO = 271;
    CITEMTYPE_ZIL = 272;
    CITEMTYPE_TUSD = 273;
    CITEMTYPE_YFI = 274;
    CITEMTYPE_MINA = 275;
    CITEMTYPE_PERP = 276;
    CITEMTYPE_XDC = 277;
    CITEMTYPE_TEL = 278;
    CITEMTYPE_SNX = 279;
    CITEMTYPE_BTG = 280;
    CITEMTYPE_AFN = 281;
    CITEMTYPE_ALL = 282;
    CITEMTYPE_AMD = 283;
    CITEMTYPE_ANG = 284;
    CITEMTYPE_AOA = 285;
    CITEMTYPE_ARS = 286;
    CITEMTYPE_AWG = 287;
    CITEMTYPE_AZN = 288;
    CITEMTYPE_BAM = 289;
    CITEMTYPE_BBD = 290;
    CITEMTYPE_BDT = 291;
    CITEMTYPE_BGN = 292;
    CITEMTYPE_BHD = 293;
    CITEMTYPE_BIF = 294;
    CITEMTYPE_BMD = 295;
    CITEMTYPE_BND = 296;
    CITEMTYPE_BOB = 297;
    CITEMTYPE_BRL = 298;
    CITEMTYPE_BSD = 299;
    CITEMTYPE_BTN = 300;
    CITEMTYPE_BWP = 301;
    CITEMTYPE_BYN = 302;
    CITEMTYPE_BZD = 303;
    CITEMTYPE_CDF = 304;
    CITEMTYPE_CLP = 305;
    CITEMTYPE_COP = 306;
    CITEMTYPE_CRC = 307;
    CITEMTYPE_CUC = 308;
    CITEMTYPE_CUP = 309;
    CITEMTYPE_CVE = 310;
    CITEMTYPE_CZK = 311;
    CITEMTYPE_DJF = 312;
    CITEMTYPE_DKK = 313;
    CITEMTYPE_DOP = 314;
    CITEMTYPE_DZD = 315;
    CITEMTYPE_EGP = 316;
    CITEMTYPE_ERN = 317;
    CITEMTYPE_ETB = 318;
    CITEMTYPE_FJD = 319;
    CITEMTYPE_FKP = 320;
    CITEMTYPE_GEL = 321;
    CITEMTYPE_GGP = 322;
    CITEMTYPE_GHS = 323;
    CITEMTYPE_GIP = 324;
    CITEMTYPE_GMD = 325;
    CITEMTYPE_GNF = 326;
    CITEMTYPE_GTQ = 327;
    CITEMTYPE_GYD = 328;
    CITEMTYPE_HNL = 329;
    CITEMTYPE_HRK = 330;
    CITEMTYPE_HTG = 331;
    CITEMTYPE_IDR = 332;
    CITEMTYPE_ILS = 333;
    CITEMTYPE_IMP = 334;
    CITEMTYPE_IQD = 335;
    CITEMTYPE_IRR = 336;
    CITEMTYPE_ISK = 337;
    CITEMTYPE_JEP = 338;
    CITEMTYPE_JMD = 339;
    CITEMTYPE_JOD = 340;
    CITEMTYPE_KES = 341;
    CITEMTYPE_KGS = 342;
    CITEMTYPE_KHR = 343;
    CITEMTYPE_KMF = 344;
    CITEMTYPE_KPW = 345;
    CITEMTYPE_KRW = 346;
    CITEMTYPE_KWD = 347;
    CITEMTYPE_KYD = 348;
    CITEMTYPE_KZT = 349;
    CITEMTYPE_LAK = 350;
    CITEMTYPE_LBP = 351;
    CITEMTYPE_LKR = 352;
    CITEMTYPE_LRD = 353;
    CITEMTYPE_LSL = 354;
    CITEMTYPE_LYD = 355;
    CITEMTYPE_MAD = 356;
    CITEMTYPE_MDL = 357;
    CITEMTYPE_MGA = 358;
    CITEMTYPE_MKD = 359;
    CITEMTYPE_MMK = 360;
    CITEMTYPE_MNT = 361;
    CITEMTYPE_MOP = 362;
    CITEMTYPE_MRU = 363;
    CITEMTYPE_MUR = 364;
    CITEMTYPE_MVR = 365;
    CITEMTYPE_MWK = 366;
    CITEMTYPE_MZN = 367;
    CITEMTYPE_NAD = 368;
    CITEMTYPE_NGN = 369;
    CITEMTYPE_NIO = 370;
    CITEMTYPE_NOK = 371;
    CITEMTYPE_NPR = 372;
    CITEMTYPE_OMR = 373;
    CITEMTYPE_PAB = 374;
    CITEMTYPE_PEN = 375;
    CITEMTYPE_PGK = 376;
    CITEMTYPE_PKR = 377;
    CITEMTYPE_PLN = 378;
    CITEMTYPE_PYG = 379;
    CITEMTYPE_QAR = 380;
    CITEMTYPE_RON = 381;
    CITEMTYPE_RSD = 382;
    CITEMTYPE_RUB = 383;
    CITEMTYPE_RWF = 384;
    CITEMTYPE_SAR = 385;
    CITEMTYPE_SBD = 386;
    CITEMTYPE_SCR = 387;
    CITEMTYPE_SDG = 388;
    CITEMTYPE_SHP = 389;
    CITEMTYPE_SLL = 390;
    CITEMTYPE_SOS = 391;
    CITEMTYPE_SPL = 392;
    CITEMTYPE_SRD = 393;
    CITEMTYPE_STN = 394;
    CITEMTYPE_SVC = 395;
    CITEMTYPE_SYP = 396;
    CITEMTYPE_SZL = 397;
    CITEMTYPE_TJS = 398;
    CITEMTYPE_TMT = 399;
    CITEMTYPE_TND = 400;
    CITEMTYPE_TOP = 401;
    CITEMTYPE_TRY = 402;
    CITEMTYPE_TTD = 403;
    CITEMTYPE_TVD = 404;
    CITEMTYPE_TWD = 405;
    CITEMTYPE_TZS = 406;
    CITEMTYPE_UAH = 407;
    CITEMTYPE_UGX = 408;
    CITEMTYPE_UYU = 409;
    CITEMTYPE_UZS = 410;
    CITEMTYPE_VEF = 411;
    CITEMTYPE_VND = 412;
    CITEMTYPE_VUV = 413;
    CITEMTYPE_WST = 414;
    CITEMTYPE_XAF = 415;
    CITEMTYPE_XCD = 416;
    CITEMTYPE_XDR = 417;
    CITEMTYPE_XOF = 418;
    CITEMTYPE_XPF = 419;
    CITEMTYPE_YER = 420;
    CITEMTYPE_ZMW = 421;
    CITEMTYPE_ZWD = 422;
    CITEMTYPE_CUSTOM = 423;
    CITEMTYPE_TNBSV = 424;
    CITEMTYPE_TNXEC = 425;
    CITEMTYPE_CSPR = 426;
    CITEMTYPE_TNCSPR = 427;
}

enum ContactItemAttribute {
    CITEMATTR_ERROR = 0;
    CITEMATTR_ACTIVE = 1;
    CITEMATTR_PRIMARY = 2;
    CITEMATTR_LOCAL = 3;
}
